### DISCUSSION TITLE
Generic Definition of local file extension when opening PDS members

### DISCUSSION THREAD

**Discussion by BJostmeyer** [IBM/zopeneditor-about#290] [Ideas]
Hi,
I like the extension for editing REXX files that are located in a Partitioned Dataset on the Host. The PDS ends with *.*.REXX(MEMBER1) and I configured the editor that members from such a PDS that ends with REXX get a (local) extension of *.rexx.

This cool feature causes, that the PDS member "MEMBER1" is downloaded as "MEMBER1.rexx" when I open it. and the syntax highlighting works like a charm :)

Now, I have a similar "problem" with JSON files that are stored in a PDS *.*.JSON(AJSON) - when I open this member, it is simply downloaded as text file - and no syntax highlighting or help for JSON files is active :(

Is it possible to have a feature in the Editor which is similar to the ones for REXX DS - but more generic? Like "Treat members of a DS with following names with a local extension of "xyz" - so then the eyntax highlighting could be enabled in this way for all kinds of files?

**Comment by phaumer** [comment]
@BJostmeyer yes, the REXX data sets feature is just a convenience method that modifies the VS Code `"files.associations"` user setting for you. To get the same effect for JSON you just have to add an entry there. So in your case add `"*.JSON*": "json"`. You can use other wildcard combinations as well that match your data sets. The `*` at the end of my example means any dataset, but you could also try patterns such as `*.DATA(JS*)`. See <https://code.visualstudio.com/docs/languages/overview#_can-i-map-additional-file-extensions-to-a-language>
